test/test_core.jl

```diff
@@ -68,6 +68,8 @@ end
             @lens _[1]
             @lens _.a.b[2]
             @lens _
+            MultiPropertyLens((a=@lens(_),))
+            (@lens _.a[1]) ∘ MultiPropertyLens((b = (@lens _[1]),))
```

Comment:
Can you add the same two tests, with `MultiPropertyLens` replaced by a `struct UserDefinedLens <: Lens`?

Reply:
Done!